From faa01696b49531688903ab87393e8a18a305faa2 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Sat, 10 May 2025 01:35:11 +0800
Subject: [PATCH 098/115] mtk: temporarily skip SSID check for non-TX BSS

mac80211 has reworked some code flow for beacon parsing since commit:
wifi: mac80211: fix MLE non-inheritance parsing

This breaks the connection between STA and 11v non-TX BSS.
If a STA connects to a non-TX BSS, it will later disconnect since the
check for SSID of non-TX BSS failed.

We are currently not sure it should be considered as "a parsing issue" or
"ieee80211_mgd_ssid_mismatch() doesn't consider 11vMBSS case before".

We may need more time to dig it out. Before this, skip beacon SSID check
for non-TX BSS.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 net/mac80211/mlme.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 73b94576..083f1438 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -7547,7 +7547,8 @@ static void ieee80211_rx_mgmt_beacon(struct ieee80211_link_data *link,
 	if (!elems)
 		return;
 
-	if (rx_status->flag & RX_FLAG_DECRYPTED &&
+	/* FIXME: temporarily skip SSID check for non-TX BSS */
+	if (rx_status->flag & RX_FLAG_DECRYPTED && !bss_conf->bss->transmitted_bss &&
 	    ieee80211_mgd_ssid_mismatch(sdata, elems)) {
 		sdata_info(sdata, "SSID mismatch for AP %pM, disconnect\n",
 			   sdata->vif.cfg.ap_addr);
-- 
2.45.2

